net/ethernet: There is no need to fill in the header in all frags

tbursztyka opened the pull request:
Only the first one requires it. Actually drivers know that already and
handle the frags list correctly.

In case ethernet has to run along with 15.4 on the same SoC, this will
optimize things quite a bit knowing that biggest ethernet frame will be
forcefully split in as many 128 bytes frags as necessary.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/ethernet: There is no need to fill in the header in all frags

Only the first one requires it. Actually drivers know that already and
handle the frags list correctly.

In case ethernet has to run along with 15.4 on the same SoC, this will
optimize things quite a bit knowing that biggest ethernet frame will be
forcefully split in as many 128 bytes frags as necessary.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

lpereira reviewed (approved):
Good catch! LGTM

Commit:
net/arp: There is no need to fill in the header in all frags

Only the first one requires it. Same issue as already fixed in ethernet
side.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

lpereira commented on subsys/net/ip/l2/ethernet/arp.c:
```diff
@@ -290,22 +289,9 @@ struct net_pkt *net_arp_prepare(struct net_pkt *pkt)
-
-		frag = frag->frags;
-	}
+	hdr = net_eth_fill_header(ctx, pkt, pkt->frags,
```
While you're cleaning this up, `hdr` doesn't seem to be used anywhere else in this function.

tbursztyka replied:
some vlan changes left over indeed.

Commit:
net/arp: Removing header filling duplicate

Header is filled in either in prepare_arp() or at the end of the
function.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
net/arp: Clear cache per-iface when relevant

When disabling an ethernet interface, only its cache entries need to be
cleared up and not the whole cache. This is meaninful in case there is
2+ ethernet interface instances.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>
